### PR TITLE
KOGITO-4442: Task Console Form breaks when task has no data assignments

### PR DESCRIPTION
**Pull request by pefernan**
The FormRenderer was breaking when the runtime provides a jsonSchema that has no properties (this happens when a task 
has no data assignments). To tackle that I added a new `EmptyTaskForm` component to display an empty state, but at same time shows the task action buttons to allow users continue with the task. At same time moved all the logic of rendering the task form and dealing with the data to a `TaskFormRenderer` component to simplify the existing `TaskForm`.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>
</details>

### CODE REVIEW

File: ui-packages/packages/task-console/server/MockData/controllers.js [hunk 1]
```diff
@@ -52,6 +61,20 @@ module.exports = controller = {
 
     const clearPhases = task.completed || task.state === 'Aborted';
 
+    if(taskWithEmptyForm.includes(task.id)) {
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/server/MockData/controllers.js]: whitespace formatting looks non-consistent here, just a proposal should we include also this code-path to be formatted automatically?

File: ui-packages/packages/task-console/src/components/Organisms/EmptyTaskForm/tests/EmptyTaskForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        formSchema={_.cloneDeep(ApplyForVisaForm)}
+        task={userTaskInstance}
+        onSubmitSuccess={formSubmitSuccessCallback}
+        onSubmitError={formSubmitErrorCallback}
+      />
+    </KogitoAppContextProvider>,
+    'EmptyTaskForm'
+  );
+
+  return wrapper.update().find(EmptyTaskForm);
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Organisms/EmptyTaskForm/tests/EmptyTaskForm.test.tsx]: Here we need to return the component including its wrappers? If so, it should be enough to call getWrapper function with EmptyTaskForm component instead of the `'EmptyTaskForm'` string as last argument.

File: ui-packages/packages/task-console/src/components/Organisms/EmptyTaskForm/tests/EmptyTaskForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+describe('EmptyTaskForm Test', () => {
+  beforeEach(() => {
+    userContext = new TestingUserContext();
+  });
+
+  it('Empty form rendering', async () => {
+    const wrapper = getEmptyTaskFormWrapper(userTaskInstance);
+
+    wrapper.update();
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Organisms/EmptyTaskForm/tests/EmptyTaskForm.test.tsx]: This should not be needed :/ What problems do you see without this?

File: ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx [hunk 4]
```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx]: date in the header

File: ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        formSchema={_.cloneDeep(ApplyForVisaForm)}
+        task={userTaskInstance}
+        onSubmitSuccess={formSubmitSuccessCallback}
+        onSubmitError={formSubmitErrorCallback}
+      />
+    </KogitoAppContextProvider>,
+    'TaskFormRenderer'
+  );
+
+  return wrapper.update().find(TaskFormRenderer);
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx]: again, not sure why is this needed here.

File: ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+  let wrapper = getTaskFormRendererWrapper(
+    userTaskInstance,
+    formSubmitSuccessCallback,
+    formSubmitErrorCallback
+  );
+
+  wrapper = wrapper.update().find(TaskFormRenderer);
+
+  expect(wrapper).toMatchSnapshot();
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Organisms/TaskFormRenderer/tests/TaskFormRenderer.test.tsx]: can we check programatically instead of snapshots? We have 5k line snapshot file as a result. If it could be replaced by a few assertions I would suggest that.

File: ui-packages/packages/task-console/src/util/uniforms/FormActionsUtils.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  if (_.isEmpty(actions)) {
+    return null;
+  }
+
+  const capitalize = label => {
+    return label.charAt(0).toUpperCase() + label.slice(1);
+  };
+
+  const isPrimary = (label: string): boolean => {
+    return label.toLowerCase() === 'complete';
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/util/uniforms/FormActionsUtils.tsx:47]: isPrimary and checking against 'complete' literal - would deserve at least a comment of why is so.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]
Just a few proposals mainly in area of testing.

File: ui-packages/packages/task-console/server/MockData/controllers.js (hunk 1 above)
Comment (pefernan) [reply ui-packages/packages/task-console/server/MockData/controllers.js]: +1 man I was expexting this to be formatted yarn format before send the PR

**Review by AjayJagan — APPROVED** [review]
LGTM :+1:

**Comment by pefernan** [comment]
@jstastny-cz could you please take a look again

**Review by jstastny-cz — APPROVED** [review]
LGTM.
